.changeset/lazy-route-modules.md

```diff
@@ -0,0 +1,62 @@
+---
+"react-router": minor
+"react-router-dom": minor
+"@remix-run/router": minor
+---
+
+**Introducing Lazy Route Modules!**
+
+In order to keep your application bundles small and support code-splitting of your routes, we've introduced a new `lazy()` route property. This is an async function that resolves the non-route-matching portions of your route definition (`loader`, `action`, `element`, `errorElement`, etc.). Additionally we've added support for route `Component` and `ErrorBoundary` fields that take precedence over `element`/`errorElement` and make a bit more sense in a statically-defined router as well as when using `route.lazy()`.
+
+Lazy routes are resolved on initial load and during the `loading` or `submitting` phase of a navigation or fetcher call. You cannot lazily define route-matching properties (`path`, `index`, `children`) since we only execute your lazy route functions after we've matched known routes.
+
+Your `lazy` functions will typically return the result of a dynamic import.
+
+```jsx
+// In this example, we assume most folks land on the homepage so we include that
+// in our critical-path bundle, but then we lazily load modules for /a and /b so
+// they don't load until the user navigates to those routes
+let routes = createRoutesFromElements(
+  <Route path="/" element={<Layout />}>
```

Comment:
Should we update all usages of `element` to `Component` in docs and release notes?

Reply:
yeah good call 👍

Reply:
I think where I tried to land was using `element` for static JSX definitions, and `Component` for static-JSON or lazily-defined.   But I could see an argument for `Component` everywhere for sure. I'd probably punt this to Michael/Ryan and see what they think?